tests/conftest.py

```diff
@@ -46,3 +47,19 @@ def foreman_api(fixture_dir, requests_mock, tmpdir):
         data = json.load(read_file)
     requests_mock.get('https://foreman.example.com/apidoc/v2.json', json=data)
     return apypie.Api(uri='https://foreman.example.com', api_version=2, apidoc_cache_dir=tmpdir.strpath)
+
+
+@pytest.fixture
+def preserve_environ():
+    old_environ = os.environ.copy()
+    try:
+        yield
+    finally:
+        os.environ = old_environ
+
+
+@pytest.mark.usefixtures("preserve_environ")
+@pytest.fixture
+def tmp_xdg_cache_home(tmpdir):
+    os.environ['XDG_CACHE_HOME'] = tmpdir.strpath
+    yield
```

Comment:
```suggestion
@pytest.mark.usefixtures("preserve_environ")
@pytest.fixture
def tmp_xdg_cache_home(tmpdir):
    os.environ['XDG_CACHE_HOME'] = tmpdir.strpath
    return tmpdir
```

That way you would use this instead of `tmpdir`. Maybe name it `tmpcache`.

Reply:
https://docs.pytest.org/en/stable/fixture.html#use-fixtures-in-classes-and-modules-with-usefixtures
...does not work with fixtures.

Reply:
thanks, done